src/components/map.js

```diff
@@ -0,0 +1,159 @@
+import { awsRegionLocations } from "../constants";
+
+const contrib = require("blessed-contrib");
+
+class Map {
+  constructor(layoutGrid, program, updateRegion) {
+    this.layoutGrid = layoutGrid;
+    this.program = program;
+    this.map = this.generateMap();
+    this.flashMarker = false;
+    this.markerLocation = program.region;
+    this.selectedLocation = undefined;
+    this.updateRegion = updateRegion;
+    this.map.key(["up"], () => {
+      this.updateSelectedPosition(false);
+    });
+    this.map.key(["down"], () => {
+      this.updateSelectedPosition(true);
+    });
+    this.map.key(["left"], () => {
+      this.updateSelectedPosition(false);
+    });
+    this.map.key(["right"], () => {
+      this.updateSelectedPosition(true);
+    });
+    this.map.key(["tab"], () => {
+      this.resetMap();
+    });
+    this.map.key(["enter"], () => {
+      if (
+        this.selectedLocation !== undefined &&
+        this.selectedLocation !== this.currentLocation
+      ) {
+        this.map.addMarker({
+          ...awsRegionLocations[this.currentLocation].coords,
+          color: "yellow",
+          char: "X",
+        });
+        this.currentLocation = this.selectedLocation;
+        // eslint-disable-next-line no-underscore-dangle
+        this.map._label.content = `Servers Location (${
+          awsRegionLocations[this.currentLocation].label
+        })`;
+        this.selectedLocation = undefined;
+        this.updateRegion(awsRegionLocations[this.currentLocation].label);
+        console.log(
+          `Welcome to ${awsRegionLocations[this.currentLocation].label}`
+        );
+      }
+    });
+  }
+
+  generateMap() {
+    const map = this.layoutGrid.set(4, 9, 4, 3, contrib.map, {
+      label: `Servers Location (${this.program.region})`,
```

Comment:
I know this was there before but "servers" is not a relevant term, let's just say location